dockerfile_parse/parser.py

```diff
@@ -100,6 +111,26 @@ def __init__(self, path=None, cache_content=False, env_replace=True, parent_env=
         else:
             self.parent_env = {}
 
+    @contextmanager
+    def _open_dockerfile(self, mode):
+        if self.fileobj is not None:
+            try:
+                self.fileobj.seek(0)
+            except IOError:
```

Comment:
If there is no real use-case with nonseekable file, than this would be the safest solution.